src/common/math/math-algebra3d.c

```diff
@@ -0,0 +1,88 @@
+#include "common/math/math-algebra3d.h"
+#include "common/math/math.h"
+#include <stdbool.h>
+#include <stdint.h>
+#include <math.h>
+
+/**
+ * Special operations in R3, as they are used often
+ * Cross product only exists in R3 
+ * Rotation matrix transpose is inverse rotation
+ */
+// vector * scalar | vector scaling
```

Comment:
I would avoid `|` since it means something specific in C. Would use `->` but that technically _also_ means something specific in C. Could be very explicit and use doxygen style, idk